UI/static/js/index.js

```diff
@@ -0,0 +1,180 @@
+    }
+
+    if (mname.value != ""){
+        if (!name_pattern.test(mname.value)){
+            mname.focus();
+            alert("Invalid middle name.\nMiddle name should only contain alpabets.\nMiddle name should not be less than three characters.");
+            return false;
+        }
+    }
+
+    if (!username_pattern.test(username.value)){
+        username.focus();
+        alert("Invalid username.\nUsername can only contain alpabets, numbers and underscore.\nUsername name should not be less than three characters.");
+        return false;
+    }
+
+    if (!email_pattern.test(email.value)){
+        email.focus();
+        alert("Invalid email.\nPlease provide a valid email");
+        return false;
+    }
+
+    if (!idNo_pattern.test(idNo.value)){
+        idNo.focus();
+        alert("Invalid ID number.\nPlease provide a valid ID Number");
+        return false;
+    }
+
+    if (!image_url_pattern.test(passport_url.value)){
+        passport_url.focus();
+        alert("Invalid passport url.\nPassport url should contain a valid domain and end with either jpg, jpeg or png.");
+        return false;
+    }
+
+    if (!phone_pattern.test(phone_number.value)){
+        phone_number.focus();
+        alert("Invalid phone number.\nPhone number should start with either '2547' or '07' followed by 8 digits. ");
+        return false;
+    }
+
+    if (!passwd_pattern.test(password.value)){
+        password.focus();
+        alert("Invalid password.\nPassword should be more than 8 characters long");
+        return false;
+    }
+
+    if (password.value != confirm_password.value){
+        confirm_password.focus();
+        alert("Passwords do not match!!!");
+        return false;
+    }
+
+    var user_data = {
+        "email": email.value,
+        "firstname": fname.value,
+        "id_no": idNo.value,
+        "lastname": lname.value,
+        "othername": mname.value,
+        "passport_url": passport_url.value,
+        "password": password.value,
+        "phone_number": phone_number.value,
+        "username": username.value
+    };
+
+    console.log(user_data);
+
+    location.replace("dashboard.html");
+
+}
+
+/**
+ * office registration
+ */
+
+function registerOffice(){
+    var office_name = document.forms.office.office_name;
+    var office_type = document.forms.office.office_type;
+
+    if (!fullname_pattern.test(office_name.value)){
+        office_name.focus();
+        alert("Invalid name.\nName should only contain alphabets and spaces if needed.\nName name should not be less than three characters.");
+        return false;
+    }
+
+    if (office_type.selectedIndex < 1){
+        office_type.focus();
+        alert("Office type is required");
+        return false;
+    }
+
+    var office_data = {
+        "type": office_type.value, 
+	    "name": office_name.value
+    };
+
+    console.log(office_data);
+
+    history.back();
+}
+
+/**
+ * party registration
+ */
+
+function registerParty(){
```

Comment:
This too can be in `js/pages/parties.js` and loaded in by the party registration page via a script tag
like

```html
<script src="./js/pages/parties.js"></script>
```

Reply:
Thank man I will look into it